internal/controller/dispatcher.go

```diff
@@ -257,3 +265,41 @@ func (r *Dispatcher) dispatch(ctx context.Context) (ctrl.Result, error) {
 
 	return ctrl.Result{RequeueAfter: dispatchDelay}, nil
 }
+
+// Calculate resource demands of appWrappers that have been dispatched but haven't
+// passed through ResourceQuota admission controller yet (approximated by resources not created yet)
+func (r *Dispatcher) getUnadmittedAppWrappersWeights(ctx context.Context) (map[string]*WeightsPair, error) {
+	appWrappers := &mcadv1beta1.AppWrapperList{}
+	if err := r.List(ctx, appWrappers, client.UnsafeDisableDeepCopy); err != nil {
+		return nil, err
+	}
+	weightsPairMap := make(map[string]*WeightsPair)
+	for _, appWrapper := range appWrappers.Items {
+		phase, step := r.getCachedAW(&appWrapper)
+		if phase == mcadv1beta1.Running &&
+			step == mcadv1beta1.Idle || step == mcadv1beta1.Creating || step == mcadv1beta1.Created {
```

Comment:
This condition does not seem right. Just use: `step != mcadv1beta1.Idle`.

Reply:
done